sdk/src/debug.rs

```diff
@@ -38,6 +38,21 @@ mod inner {
         }
     }
 
+    /// Saves an artifact to the host env. New artifacts with the same name will overwrite old ones
+    pub fn capture_artifact(name: impl AsRef<str>, data: Vec<u8>) {
```

Comment:
```suggestion
    pub fn capture_artifact(name: impl AsRef<str>, data: impl AsRef<[u8]>) {
```

Or just `&[u8]` is probably sufficient. But we definitely don't need ownership.